homework week3 Ali Amouri Kadhim

AliKdhim87 opened the pull request.

AliKdhim87 committed:
finished project for homework week3

nudge commented on Week3/homework/addToShoppingCart.js:
```diff
@@ -0,0 +1,18 @@
+'use strict';
+function addToShoppingCart(item) {
+    // An array with two Items
+    var array = ['bananas', 'milk'];
```
1. Use `let`
2. Use a more descriptive name than `array`, perhaps `shoppingCart` works better.

AliKdhim87 replied:
Thank you for the feed back I fixed

nudge commented on Week3/homework/addToShoppingCart.js:
```diff
@@ -0,0 +1,18 @@
+}
+console.log(addToShoppingCart('Tomato')); //Output You bought milk,Tomato !
+console.log(addToShoppingCart('Water')); //You bought milk,Water ! 
+console.log(addToShoppingCart('Orange')); //You bought milk,Orange !
```
The question is asking that you can hold maximum 3 items in the shopping cart, so the expected output should be:
```
You bought bananas,milk,Tomato !
You bought milk,Tomato,Water !
You bought Tomato,Water,Orange !
```
See if you can update this code to fix this. Hint: Update the check of the array length!

AliKdhim87 replied:
Thank you for the feed back I fixed

nudge commented on Week3/homework/calculateDogAge.js:
```diff
@@ -0,0 +1,8 @@
+'use strict';
+function calculateDogAge(puppysAge) {
+    const dogYears = 7 * puppysAge;
+    return console.log(`Your doggie is${dogYears} years old in dog years!`);
```
The question asks to return a string here.

See if you can update the code to fix this. Hint: Move the `console.log` outside this function.

AliKdhim87 replied:
Thank you for the feed back I fixed.

nudge commented on Week3/homework/calculateTotalPrice.js:
```diff
@@ -0,0 +1,18 @@
+'use strict';
+const calculateTotalPrice = (object) => {
+    let sum = null;
```
nit: might be clearer to do `let sum = 0;`

AliKdhim87 replied:
Thank you for the feedback I fixed

nudge commented on Week3/homework/calculateTotalPrice.js:
```diff
@@ -0,0 +1,18 @@
+    water: 0.9,
+    chhocolateBar: 1.75
+};
+console.log(calculateTotalPrice(cartForParty));
```
👍

AliKdhim87 replied:
Thank you for the feedback

nudge commented on Week3/homework/giveCompliment.js:
```diff
@@ -0,0 +1,12 @@
+    // Array with ten Complimen
+    const complimentArray = ['great', 'awesome', 'great', 'nice', 'Elegant', 'Elegant', 'Majestic', 'Lovely', 'Striking', 'Amazing'];
+    // to ronden the content of the array 
+    const randomArray = Math.floor(Math.random() * complimentArray.length);
```
nit: `randomIndex` might be a clearer name

AliKdhim87 replied:
Thank you for the feed back I changed the name

nudge commented on Week3/homework/giveCompliment.js:
```diff
@@ -0,0 +1,12 @@
+}
+console.log(giveCompliment('Ali'));
+console.log(giveCompliment('Ali'));
+console.log(giveCompliment('Ali'));
```
👍

AliKdhim87 replied:
Thank you for the feedback

nudge commented on Week3/homework/tellFortune.js:
```diff
@@ -0,0 +1,15 @@
+const locations = ['Irak', 'Netherland', 'Brazil', 'Gernany', 'Turkey'];
+const jobs = ['Front-End developer', 'Back-end developer', 'Full-Stack developer', 'Sport coach', 'Policeman'];
+
+tellFortune(numChildren, partnerNames, locations, jobs);
```
👍  nice! Great use of string templates.

AliKdhim87 replied:
Thank you very much for the feedback

nudge commented on Week3/homework/creditCardValidator.js:
```diff
@@ -0,0 +1,54 @@
+        if (!Number.isInteger(currentNumber)) {
+            return false;
+        }
+    }
```
Your logic works which is great.
Another way to do this more compactly is using `Number()` object, see: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number
It tries to parse the entire string and see if it is a number, and if not returns `NaN`.

You could do a check like `Number.isInteger(Number(creditCardNum))`

nudge commented on Week3/homework/creditCardValidator.js:
```diff
@@ -0,0 +1,54 @@
+console.log(validateCreditCard('6666666666661666')); //true
+console.log(validateCreditCard('a92332119c011112')); //false
+console.log(validateCreditCard('4444444444444444')); //false
+console.log(validateCreditCard('1211111111111112')); //true
```
👍 nicely done!!

AliKdhim87 committed:
thank you for the feedback  

I changed the var to let and about the amount of the items in the array i just follow the construction of the  exercise

AliKdhim87 committed:
thank you for the feedback

I changed

AliKdhim87 committed:
thank you for the feedback

I changed the value of the let sum from null to 0

AliKdhim87 committed:
thank you for the feedback

I changed to  this one Number.isInteger(Number(creditCardNum))

AliKdhim87 committed:
Thank you for the feedback 

I changed the name of the variable

AliKdhim87 reviewed (commented):
I resolved all this problem.
